server/src/service-layer/controllers/AdminController.ts

```diff
@@ -329,4 +332,41 @@ export class AdminController extends Controller {
       this.setStatus(500) // unknown server error?
     }
   }
+
+  @SuccessResponse("200", "Coupon Added")
+  @Post("users/add-coupon")
+  public async addCoupon(
+    @Body() requestBody: AddCouponRequestBody
+  ): Promise<void> {
+    const { uid, quantity } = requestBody
+    const amount = 40 // Hardcoded amount
+    try {
+      const userService = new UserDataService()
+      const stripeService = new StripeService()
+      const user = await userService.getUserData(uid)
+
+      if (!user) {
+        this.setStatus(404)
+        return
+      }
+      if (!user.stripe_id) {
+        this.setStatus(400)
+        return
+      }
+
+      // Add coupon to the user using Stripe ID
+      for (let i = 0; i < quantity; i++) {
+        await stripeService.addCouponToUser(user.stripe_id, amount)
+      }
+
+      const couponPromises = Array.from({ length: quantity }, () =>
```

Comment:
This isn't a promise, need to make it async
```suggestion
      const couponPromises = Array.from({ length: quantity }, async () =>
```